007-design-docs.md

```diff
@@ -0,0 +1,97 @@
+
+### How to write a design document
+
+The design documents would live in the strimzi-kafka-operator repo and would match the current code state, so that code changes could be audited with their corresponding design doc changes.
+
+A design document would be split into `X` parts
+
+**Title**
+
+The title of the design doc, normally this would pertain to a single component or one larger aspect of the whole of the Strimzi project. New pages would be discussed and decided on as part of a proposal where needed.
+
+**Overview**
+
+A high level summary that any user of Strimzi should be able understand and use to decide if it’s useful for them to read the rest of the doc. It should be 3 paragraphs max.
+
+**Context**
+
+A description of the problem the program/code tackles, why the program/code is necessary, what people need to know to assess this project, and how it fits into the technical strategy of Strimzi.
+
+**Goals and Non-Goals**
+
+The Goals section should describe the user-driven impact of the code and specify how to measure success using metrics, normally that would be tested.
+Non-Goals are equally important to describe which problems you won’t be fixing so everyone is on the same page.
+
+**Milestones**
+
+A list of measurable checkpoints, if there is future progress planned, including a milestone for the future in the design doc maye be useful for a reader to understand where the project is heading.
+
+**High Level Description**
+
+High level description describing the current implementation as well as example flow(s) to illustrate how users interact with this system and/or how data flow through it. This should build on the context.
+
+**Technical Architecture section**
+
+Walk through the flows in great detail to concertise the implementation. Feel free to include diagrams where relevant and sub-sections for edge-cases and implementation details.
+
+## Summary of changes
+
+Alongside code changes, behavioural changes of the operator will also be documented in the above described design documentation format.
+
+Newly contributed proposals or issues would describe how their changes might alter the design documentation and are expected to be delivered alongside their code changes.
+
+Design documents accumulate and form a body of design information for the project, and should be kept up to date with design changes. 
+
+
+
+## Alternative implementation - Strimzi Improvement Proposal (SIP)
+
+Inspired and closely resembling [Kafka Improvement Proposals (KIP)](https://cwiki.apache.org/confluence/display/KAFKA/Kafka+Improvement+Proposals), we merge the concept of design documentation and proposals together, meaning that each proposal is a point in time descriptor of current Strimzi behaviour and a proposal as to how it should change.
```

Comment:
As described in other comments. I think tech design docs and proposals are two very different things. They should (in the right format) ideally coexist next to each other.

Reply:
While I agree that there are two purposes to be served, I think by making there be two separate processes we're in danger of requiring more and more work to be done to actually change anything. That's not to say that there aren't benefits to this approach, but I'd rather we erred on the side of having lighter weight processes and requirements until such time as we've shown that such an approach doesn't work.